src/toast.toasts.ts

```diff
@@ -0,0 +1,352 @@
+import './icon-button.js';
+import { html, LitElement } from 'lit';
+import { createRef, ref } from 'lit/directives/ref.js';
+import { customElement, property } from 'lit/decorators.js';
+import { repeat } from 'lit/directives/repeat.js';
+import { classMap } from 'lit/directives/class-map.js';
+import { styleMap } from 'lit/directives/style-map.js';
+import { choose } from 'lit/directives/choose.js';
+import { when } from 'lit/directives/when.js';
+import { unsafeHTML } from 'lit/directives/unsafe-html.js';
+import xIcon from './icons/x.js';
+import { LocalizeController } from './library/localize.js';
+import styles from './toast.toasts.styles.js';
+import shadowRootMode from './library/shadow-root-mode.js';
+import final from './library/final.js';
+import GlideCoreToast from './toast.js';
+import GlideCoreLink from './link.js';
+
+declare global {
+  interface HTMLElementTagNameMap {
+    'glide-core-private-toasts': GlideCoreToasts;
+  }
+}
+
+/**
+ * @attr {GlideCoreToast[]} [toasts=[]]
+ *
+ * @method showPopover
+ */
+@customElement('glide-core-private-toasts')
+@final
+export default class GlideCoreToasts extends LitElement {
+  static override shadowRootOptions: ShadowRootInit = {
+    ...LitElement.shadowRootOptions,
+    mode: shadowRootMode,
+  };
+
+  static override styles = styles;
+
+  @property({ type: Array })
+  toasts: GlideCoreToast[] = [];
+
+  override firstUpdated() {
+    if (this.#componentElementRef.value) {
+      this.#componentElementRef.value.popover = 'manual';
+    }
+  }
+
+  override render() {
+    // Lit-ally doesn't know that our "mouseover" listener is only an affordance for
+    // mouse users. It also doesn't know that the purpose of the "click" listener on
+    // the "description" slot is only so we can programmatically click the consumer's
+    // Toast.
+    //
+    /* eslint-disable lit-a11y/click-events-have-key-events, lit-a11y/mouse-events-have-key-events */
+    return html`<div
+      aria-label=${this.#localize.term('notifications')}
+      class="component"
+      data-test="component"
+      role="region"
+      tabindex="-1"
+      ${ref(this.#componentElementRef)}
+    >
+      <div class="toasts">
+        ${repeat(
+          this.toasts,
+          (toast) => toast.id,
+          (toast) => {
+            return html`<div
+              aria-labelledby="prefix label description"
+              class=${classMap({
+                toast: true,
+                error: toast.variant === 'error',
+                informational: toast.variant === 'informational',
+                success: toast.variant === 'success',
+                show: Boolean(toast.privateShow),
+                hide: Boolean(toast.privateHide),
+                'via-dismiss-button': Boolean(
```

Comment:
Any naming suggestions? Design wants toasts to transition [differently](https://github.com/CrowdStrike/glide-core/pull/847/files#diff-8aafe9cadefb9c2f52762c8bd9774ee153d8bcead8e7db80c465355d7cce9f25R84) when dismissed via the dismiss button.